docs/features/object-pooling/object-pooling.md

```diff
@@ -0,0 +1,177 @@
+---
+id: object-pooling
+title: Object Pooling
+sidebar_label: Object Pooling
+---
+
+Object pooling is an optimization pattern. It's used to improve performance, in certain cases, by re-using objects instead of allocating memory for them on demand. In C/C++, one the things object pooling has to offer is a solution to avoid [memory fragmentation](http://stackoverflow.com/questions/3770457/what-is-memory-fragmentation). In C#, we don't have to worry about memory fragmentation thanks to [garbage collection](https://msdn.microsoft.com/en-us/library/ee787088). However, garbage collection can be still be too expensive for certain parts of real-time applications, especially on mobile devices with slower CPUs and simpler garbage collectors. [More details on object pooling here](http://gameprogrammingpatterns.com/object-pool.html).
+
+(Abstract analogy) You can think about an object pool like desks in a classroom.  There are a designated number of desks in each room for students to use.  When a student takes over a desk for the class period, their information is assigned to that spot, and they place their items in the desk.  When the student leaves for their next class, the desk still remains (_The pool still has the object_).  When a student arrives for the next class, the old desk isn't tossed in the trash (_Garbage collected_), nor is a new desk created (_Class instantiation_), and then installed in the class (_stored in a list_).  Instead the new student simply uses one of the open desks (_object reuse/pooling_).  This saves on time, just like with Object Pooling.
```

Comment:
Could you make this section a tip admonition? You can specific `Abstarct Analogy` as the title.

The syntax for it is 

```md
:::note[Abstract Analogy]  

You can think about an object pool like desks in a classroom.  There are a designated number of desks in each room for students to use.  When a student takes over a desk for the class period, their information is assigned to that spot, and they place their items in the desk.  When the student leaves for their next class, the desk still remains (_The pool still has the object_).  When a student arrives for the next class, the old desk isn't tossed in the trash (_Garbage collected_), nor is a new desk created (_Class instantiation_), and then installed in the class (_stored in a list_).  Instead the new student simply uses one of the open desks (_object reuse/pooling_).  This saves on time, just like with Object Pooling.  

:::
```

Reply:
On neat, I'm new to Docusaurus so these tips are a great help thank you!